### PR TITLE
Unit test/realtime account validation

### PR DESCRIPTION
**Pull request by vnsrz**
Adds unit tests for the file `realtime_account_validation.js`, particularly on the password validation.

This was made for educational purposes.

### Technical

To test the JQuery in the `initRealtimeValidation` method, a `setup-jest.js` file importing the `$` identier had to be created and added to the `package.json`. As far as I know, this doesn't affect the other tests being ran, as they still passed.

### Testing

Run `docker compose exec web make test`, there should be 17 tests running and passing now, as opposed to 16 before.

### Screenshot

Screenshot_88

### CODE REVIEW

**Comment by cclauss** [comment]
```
eslint...................................................................Failed
- hook id: eslint
- exit code: 1

/code/scripts/solr_restarter/index.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

/code/tests/screenshots/index.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

/code/tests/setup-jest.js
  2:1   error  'global' is not defined  no-undef
  2:12  error  'global' is not defined  no-undef

/code/.storybook/preview.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

/code/vendor/js/jquery-form/jquery.form.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

✖ 6 problems (2 errors, 4 warnings)

/code/.storybook/main.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

/code/conf/svgo.config.js
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override

✖ 2 problems (0 errors, 2 warnings)
```

**Comment by cclauss** [comment]
```
> npx concurrently --group npm:lint:js npm:lint:css

[lint:js] 
[lint:js] > openlibrary@1.0.0 lint:js
[lint:js] > eslint --ext js,vue .
[lint:js] 
[lint:js] 
[lint:js] /home/runner/work/openlibrary/openlibrary/tests/setup-jest.js
[lint:js]   2:1   error  'global' is not defined  no-undef
[lint:js]   2:12  error  'global' is not defined  no-undef
[lint:js] 
[lint:js] ✖ 2 problems (2 errors, 0 warnings)
[lint:js] 
[lint:js] npm run lint:js exited with code 1
[lint:css] 
[lint:css] > openlibrary@1.0.0 lint:css
[lint:css] > stylelint ./**/*.less
[lint:css] 
[lint:css] npm run lint:css exited with code 0
Error: Process completed with exit code 1.
```

**Comment by vnsrz** [comment]
> ```
> > npx concurrently --group npm:lint:js npm:lint:css
> 
> [lint:js] 
> [lint:js] > openlibrary@1.0.0 lint:js
> [lint:js] > eslint --ext js,vue .
> [lint:js] 
> [lint:js] 
> [lint:js] /home/runner/work/openlibrary/openlibrary/tests/setup-jest.js
> [lint:js]   2:1   error  'global' is not defined  no-undef
> [lint:js]   2:12  error  'global' is not defined  no-undef
> [lint:js] 
> [lint:js] ✖ 2 problems (2 errors, 0 warnings)
> [lint:js] 
> [lint:js] npm run lint:js exited with code 1
> [lint:css] 
> [lint:css] > openlibrary@1.0.0 lint:css
> [lint:css] > stylelint ./**/*.less
> [lint:css] 
> [lint:css] npm run lint:css exited with code 0
> Error: Process completed with exit code 1.
> ```

Fixed

**Comment by jimchamp** [comment]
Blocked by changes to the registration page: #7670.

File: package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "workbox-webpack-plugin": "6.1.5"
   },
   "jest": {
+    "setupFiles": ["./tests/unit/setup-jest.js"],
```
Comment (jimchamp) [inline package.json:100]:
```suggestion
```
JQuery was globally exposed to Jest in #7962, so this is no longer needed.

**Review by jimchamp — APPROVED** [review]
Sorry for the delay, @vnsrz. This looks good to me.  I'm removing the Jest setup steps that you have added, as similar code was added by #7962.

Some of the tests could be simplified with a second `beforeEach` call that sets the variables that are same and used by each test.  I'll make some small changes to DRY the tests in another PR.  Will mention you as a stakeholder so that you'll see what I mean.

**Pull request merged**